vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py

```diff
@@ -0,0 +1,264 @@
+    # end_date = site["attributes"].get("end_date")
+
+    status = site["attributes"].get("status")
+
+    status_options = {
+        "Open": True,
+        "Closed": False,
+        "Testing Restricted": True,
+        "Scheduled to Open": False,
+        "Temporarily Closed": False,
+    }
+
+    return try_lookup(status_options, status, None, name="active status lookup")
+
+
+def _get_access(site: dict) -> Optional[List[str]]:
+    drive = site["attributes"].get("drive_through")
+    drive_bool = drive is not None
+
+    # walk = site["attributes"].get("drive_through")
+    # walk_bool = drive is not None
+
+    wheelchair = site["attributes"].get("Wheelchair_Accessible")
+
+    wheelchair_options = {
+        "Yes": "yes",
+        "Partially": "partial",
+        "Unknown": "no",
+        "Not Applicable": "no",
+        "NA": "no",
+    }
+    wheelchair_bool = try_lookup(
+        wheelchair_options, wheelchair, "no", name="wheelchair access"
+    )
+
+    return schema.Access(drive=drive_bool, wheelchair=wheelchair_bool)
+
+
+def try_lookup(mapping, value, default, name=None):
+    if value is None:
+        return default
+
+    try:
+        return mapping[value]
+    except KeyError as e:
+        name = " for " + name or ""
+        logger.error("value not present in lookup table%s: %s", name, e)
+
+        return default
+
+
+def _get_published_at(site: dict) -> Optional[str]:
+    date_with_millis = site["attributes"]["CreationDate"]
+    if date_with_millis:
+        date = datetime.datetime.fromtimestamp(date_with_millis / 1000)  # Drop millis
+        return date.isoformat()
+
+    return None
+
+
+def try_get_list(lis, index, default=None):
+    if lis is None:
+        return default
+
+    try:
+        value = lis[index]
+        if value == "none":
+            logger.warn("saw none value")
+        return value
+    except IndexError:
+        return default
+
+
+def _get_normalized_location(site: dict, timestamp: str) -> schema.NormalizedLocation:
+
+    addrsplit = site["attributes"]["fulladdr"].split(", ")
+
+    zip = site["attributes"]["fulladdr"][-5:]
+    zip = zip if zip.isnumeric() else None
+
+    city_state_zip = addrsplit[1].split(" ") if try_get_list(addrsplit, 1) else None
+
+    state = site["attributes"]["State"] or None
+    state = state.strip() if state is not None else None
+
+    return schema.NormalizedLocation(
+        id=f"{SOURCE_NAME}:{_get_id(site)}",
+        name=site["attributes"]["name"],
+        address=schema.Address(
+            street1=addrsplit[0],
+            street2=None,
+            city=site["attributes"]["municipality"]
+            or try_get_list(city_state_zip, -3, default=""),
+            state=state,
+            zip=zip,
+        ),
+        location=schema.LatLng(
+            latitude=site["geometry"]["y"] if site.get("geometry") else None,
+            longitude=site["geometry"]["x"] if site.get("geometry") else None,
+        )
```

Comment:
`latitude` and `longitude` are both required if a `LatLng` is set - do we need the `if`s in here?

I see the note below to ask about how it's convoluted 🙃.

Since this is scraped from an ArcGIS source, I'm also not too concerned about these being null, so going to ignore this before merging.

Reply:
yeah, essentially the line after it is supposed to basically make the whole thing null if the "geometry" key is not set, but looking at this again, a try/catch may be better here